clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -929,8 +943,9 @@ private void throwIfProducerClosed() {
             if (log.isTraceEnabled()) {
                 log.trace("Attempting to append record {} with callback {} to topic {} partition {}", record, callback, record.topic(), partition);
             }
+            msgQueuingTime.set(time.milliseconds());
```

Comment:
this msgQueuingTime has to be local instead of member variable since kafka producer is thread safe, i.e. multiple threads can call send at the same time. We need to ensure this end-to-end time is for the correct sender.

Also, the start time should be in the very beginning, before the waitOnMetadata since the sensor is end-to-end latency, which should include time spent waiting on md.